packages/framework/client-logger/app-insights-logger/src/test/fluidAppInsightsLogger.spec.ts

```diff
@@ -90,9 +142,29 @@ describe("Telemetry Filter - filter mode", () => {
 	});
 });
 
-describe("Telemetry Filter - Category filtering", () => {
+describe("Telemetry Filter - Category Filtering", () => {
 	let appInsightsClient: ApplicationInsights;
 	let trackEventSpy: Sinon.SinonSpy;
+	const configFilters: TelemetryFilter[] = [
+		{
+			categories: ["performance"],
+		},
+		{
+			categories: ["generic"],
+		},
+	];
```

Comment:
Updated the code to include a validation function for filters which:
1. merges redundant pure category filters (and console.warns)
2. throws error if duplicate namespace filters are provided
3. throws error if namespace pattern exception is provided that is not a child of the parent namespace

Reply:
Per our other conversation, this was updated to throw an error if you define multiple filters  that only define categories rather than merge them internally.